### PR TITLE
eval_metrics

### PR DESCRIPTION
**Pull request by J22Melody**
feat(pose_to_segments): add evaluation metrics and ensure data correctness

### CODE REVIEW

File: pose_to_segments/src/data.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def build_bio(identifier: str, timestamps: torch.Tensor, segments: List[Segment]
         while timestamps[timestamp_i] < segment["start_time"]:
             timestamp_i += 1
         segment_start_i = timestamp_i
-        while timestamp_i < len(timestamps) and timestamps[timestamp_i] < segment["end_time"]:
+        while timestamp_i < (len(timestamps) - 1) and timestamps[timestamp_i] < segment["end_time"]:
```
Comment (J22Melody) [inline pose_to_segments/src/data.py:57]: @AmitMY this should fix a index bug

File: pose_to_segments/src/utils/metrics.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import torch
+import numpy as np
+
+
+def frame_accuracy(probs, gold):
```
Comment (AmitMY) [inline pose_to_segments/src/utils/metrics.py:5]: can you please:
1. in a comment, add the expected shape of `probs` and `gold`?
2. add python typing (`frame_accuracy(probs: torch.Tensor....`)

File: pose_to_segments/src/train.py [hunk 3]
```diff
@@ -23,10 +23,15 @@
...
                      hand_normalization=args.hand_normalization,
-                     optical_flow=args.optical_flow)
+                     optical_flow=args.optical_flow,
+                     data_dir=args.data_dir)
 
-    train_dataset = get_dataset(split="train", **data_args)
-    train_loader = DataLoader(train_dataset, batch_size=args.batch_size, shuffle=True, collate_fn=zero_pad_collator)
+    if args.data_dev:
+            train_dataset = get_dataset(split="validation", **data_args)
+            train_loader = DataLoader(train_dataset, batch_size=args.batch_size, shuffle=False, collate_fn=zero_pad_collator)
```
Comment (AmitMY) [inline pose_to_segments/src/train.py:31]: indentation looks wrong

File: pose_to_segments/src/model.py [hunk 4]
```diff
@@ -67,26 +71,60 @@ def training_step(self, batch, *unused_args):
...
+
+            # assign masked postions the O tag
+            probs_masked = probs.detach().clone()
+            probs_masked[(mask == 0).nonzero(as_tuple=True)[0]] = torch.log(torch.tensor([1, 0, 0]).cuda())
+
+            metrics['frame_accuracy'].append(frame_accuracy(probs_masked, gold))
+
+            segments = probs_to_segments(probs_masked.cpu())
+            # convert segments from second to frame
+            fps = 25
```
Comment (AmitMY) [inline pose_to_segments/src/model.py:97]: `fps` is actually a parameter in the data. looking at `batch["pose"]["obj"].body.fps` should give you the number as well

**Review by AmitMY — CHANGES_REQUESTED** [review]

**Review by AmitMY — APPROVED** [review]

**Pull request merged**